setup-r-dependencies/action.yaml

```
@@ -116,24 +119,63 @@ runs:
         shell: Rscript {0}
         working-directory: ${{ inputs.working-directory }}
 
-      - name: Restore R package cache
-        if: inputs.cache == 'true'
+      - name: R package cache
+        if: inputs.cache != 'false'
         uses: actions/cache@v4
         with:
           path: |
             ${{ env.R_LIBS_USER }}/*
             !${{ env.R_LIBS_USER }}/pak
           key: ${{ format('{0}-{1}-{2}-{3}', steps.install.outputs.os-version, steps.install.outputs.r-version, inputs.cache-version, hashFiles(format('{0}/.github/pkg.lock', inputs.working-directory ))) }}
-          restore-keys: ${{ steps.install.outputs.os-version }}-${{ steps.install.outputs.r-version }}-${{inputs.cache-version }}-
+          restore-keys: ${{ format('{0}-{1}-{2}-', steps.install.outputs.os-version, steps.install.outputs.r-version, inputs.cache-version) }}
+          # Used to save package installation on troublesome Actions
+          save-always: ${{ inputs.cache == 'always' }}
 
       - name: Install dependencies
+        id: install-pkgs
         run: |
           # Install/Update packages
           cat("::group::Install/update packages\n")
           Sys.setenv("PKGCACHE_HTTP_VERSION" = "2")
           library(pak, lib.loc = Sys.getenv("R_LIB_FOR_PAK"))
-          pak::lockfile_install(".github/pkg.lock")
-          ## Clean up lock file
+          pkgs <- pak::lockfile_install(".github/pkg.lock")
+          # Store the installed packages
+          packages <- paste0(pkgs$package, collapse = "|")
+          cat("packages=", packages, "\n", file = Sys.getenv("GITHUB_OUTPUT"), sep = "", append = TRUE)
+          cat("::endgroup::\n")
+        shell: Rscript {0}
+        working-directory: ${{ inputs.working-directory }}
+
+      - name: Prune unused R packages
+        if: inputs.cache == 'always' && inputs.prune == 'true'
```

@gaborcsardi I believe this should be done independently of `inputs.cache`. From a reproducibility standpoint, the install environment should be consistent over time and pruning helps achieve this by removing packages that are only found in the cache.

```suggestion
        if: inputs.prune == 'true'
```